src/server/api/user.js

```diff
@@ -96,5 +99,59 @@ export default function userRouter() {
         })
     );
 
+    logger.info("adding the GET /webhooks/types handler");
+    api.get(
+        "/webhooks/types",
+        auth.authnTokenMiddleware,
+        terrainHandler({
+            method: "GET",
+            pathname: "/webhooks/types",
+        })
+    );
+
+    logger.info("adding the GET /webhooks/topics handler");
+    api.get(
+        "/webhooks/topics",
+        auth.authnTokenMiddleware,
+        terrainHandler({
+            method: "GET",
+            pathname: "/webhooks/topics",
+        })
+    );
+
+    logger.info("adding the PUT /webhooks handler");
+    api.put(
+        "/webhooks",
+        auth.authnTokenMiddleware,
+        terrainHandler({
+            method: "PUT",
+            pathname: "/webhooks",
+            headers: {
+                "Content-Type": "application/json",
+            },
+        })
+    );
+
+    api.get("/testWebhook", function (req, res) {
+        axiosInstance
+            .request({
+                method: "POST",
+                url: req.query.url,
+                data: WEBHOOK_TEST_BODY,
+            })
+            .then((apiResponse) => {
+                res.set(apiResponse.headers);
+                res.status(apiResponse.status);
+                res.send(apiResponse.data);
+            })
+            .catch(async (err) => {
+                logger.error(err);
+                const e = await err;
+
+                res.status(e.response?.status || 500);
+                res.send(e.response?.data);
+            });
+    });
```

Comment:
This is a copy-pasta from `/server/external.js` since I couldn't figure out how to pass the test url by query param to `external.js` yet.

Reply:
I'm not sure either, but it looks like the external API handler is not setup to handle `POST` requests with a `body` anyway (though I'm sure it could be modified to accept a `body` param if needed).